Bio/EBI/dbfetch.py

```diff
@@ -0,0 +1,82 @@
+# This code is part of the Biopython distribution and governed by its
+# license.  Please see the LICENSE file that should have been included
+# as part of this package.
+
+"""Provides code to access the EBI dbfetch tool.
+
+This module aims to make dbfetch easy to use. See:
+http://www.ebi.ac.uk/Tools/dbfetch/
+
+The dbfetch tool from EBI provvides access to different databases via a simple
+URL that the module constructs. Errors are handled regarding databse access and
+HTPP level.
+
+The functionality is somewhat similar to Biopython's Bio.TogoWS, Bio.KEGG.REST
+and Bio.Entrez modules.
+
+EBI provvides extensive guidelines for the databases, their format and the
+syntax necessary to use the tool. See:
+http://www.ebi.ac.uk/Tools/dbfetch/syntax.jsp
+
+References:
+Lopez R, Cowley A, Li W, McWilliam H.;
+(2013) Using EMBL-EBI Services via Web Interface and Programmatically via
+Web Services. Curr Protoc Bioinformatics Volume 48 (2014) p.3.12.1-3.12.50.
+
+McWilliam H., Li W., Uludag M., Squizzato S., Park Y.M., Buso N., Cowley A.P.,
+Lopez R.;(2013) Analysis Tool Web Services from the EMBL-EBI Nucleic Acids
+Research 41: W597-W600.
+
+"""
+
+from __future__ import print_function
+
+from Bio._py3k import urlopen as _urlopen
+from Bio._py3k import _binary_to_string_handle
+
+import requires_internet
+requires_internet.check()
+
+
+def _q(arg1, arg2=None, arg3=None, arg4=None):
```

Comment:
The argument names: arg1, arg2, arg3, arg4 is hard to understand the behavior.
The reason you used those may be that you wanted to call `_q` from both by `def dbfetch` and `def db_info`.
Is this case, replacing `def _q` to `def _query_by_db`, below sample looks better, isn't it?

```
_BASE_URL = "http://www.ebi.ac.uk/Tools/dbfetch/dbfetch%s"


def _query_by_url(url):
    resp = _urlopen(_BASE_URL % url)
    return _binary_to_string_handle(resp)


def db_info():
    return _query_by_url("dbfetch.databases?style=json")


def dbfetch(db, ids, format=None, style=None):
    ...

    return _query_by_db(db, ids, format, style)


def _query_by_db(db, ids=None, format=None, style=None):
    url = '/%s' % db
    if ids:
        url += '/%s' % ids
    if format:
        url += '/%s' % format
    if style:
        url += '/%s' % style
    resp = _urlopen(_BASE_URL % url)
    return _binary_to_string_handle(resp)
```